pkg/registry/generic/etcd/etcd.go

```diff
@@ -131,6 +135,26 @@ func NamespaceKeyFunc(ctx api.Context, prefix string, name string) (string, erro
 	return key, nil
 }
 
+// Default function for removing all resources of a type.
+func DeleteCollectionPerNamespaceFunc() func(api.Context, storage.Interface, string) (runtime.Object, error) {
+	return func(ctx api.Context, s storage.Interface, prefix string) (runtime.Object, error) {
+		ns, ok := api.NamespaceFrom(ctx)
+		if !ok || len(ns) == 0 {
+			return nil, kubeerr.NewBadRequest("Namespace parameter required.")
+		}
+		var out runtime.Object
```

Comment:
You should pass NewListFunc (from Etcd) to this function and use:
"out := NewListFunc()" 
instead.